draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,551 @@ author:
+
+2) If either the authority hostname or any of the hostnames returned
+   in the Proxy-DNS-Used list match the TargetName of the selected
+   alternative endpoint, and if the port used in the CONNECT* matches
+   the selected alternative endpoint, and if the transport protocol is
+   compatible with the selected alpn (eg, CONNECT for "h2" and
+   CONNECT-UDP for "h3"), the client SHOULD use the established
+   connection.
+
+3) If no alternative endpoint has a match with either condition 1
+   or 2, the client MUST NOT use the connection and will instead
+   abandon it.  Clients then MUST establish a new CONNECT*
+   based on the TargetName and other SvcParams from the selected
+   alternative endpoint.
+
+Note that clients MAY select to use a less preferred alternative
+endpoint (if only temporarily) if it allows use of the
+opportunistically established connection, but only provided
+either condition 1 or 2 matches.
+
+If a CONNECT* request does not include a Proxy-DNS-SVCB header, and if
+clients do not have a cached SVCB RR, SVCB-optional clients SHOULD
+proceed with using the established connection.
+
+For all of the above, clients MUST age out information learned
+about Proxy-DNS-Used and Proxy-DNS-SVCB based on the TTLs
+returned in those headers.  Clients SHOULD continue to refresh
+them as requests are made to the proxy.  Clients SHOULD periodically
+re-evaluate if new connections need to be established based on
+expiry of these TTLs.
+
+*TODO*: Most of this section assumes ServiceMode records.
+        We should better structure to handle AliasMode records.
+        (I'm not sure the second condition ever applies?)
+        Below is some partial text:
+
+When Proxy-DNS-SVCB (or a cached SVCB) contains an AliasMode record,
+clients SHOULD either use a connection made to the TargetName
+of the AliasMode record, or they SHOULD use a connection
+where any of the hostnames returned in Proxy-DNS-Used
+matches the TargetName of the AliasMode record.
+
+
+## SVCB-required clients
+
+SVCB-required clients must either perform additional SVCB
+DNS requests, or MUST use a proxy that is known and configured
+to always return Proxy-DNS-SVCB and Proxy-DNS-Used headers.
+
+SVCB-required clients SHOULD set high values for "wait" in
+Proxy-DNS-Request and MUST treat the absence of a Proxy-DNS-SVCB
+response header as an error.
+
+SVCB-required clients may also need to make their initial request to
+an authority with no expectation of being able to use that connection.
+Protocols specifying for SVCB-required clients will need to describe
+what clients should use in this case.
+
+
+## Connection coalescing
+
+*TODO*: Do we want to include text on how clients
+can also use Proxy-DNS-Used for connection coalescing?
+(One thing missing there is the full set of IP addresses
+to look for overlaps within an address family.   I'm not
+sure this is worth the complexity.  Or do we remove this
+sub-section entirely?)
```

Comment:
My vote is for removal.  Coalescing based on IP address is not a battle we've been winning, and I don't think this draft is the venue to continue fighting it.